### ISSUE TITLE
Windows 365 Enterprise

### ISSUE THREAD

**Issue by AWeber78** [microsoft/Microsoft365DSC#4158]
### Description of the issue

Windows 365 Enterprise Settings are missing in Microsoft365DSC

### Microsoft 365 DSC Version

1.24.110.1

### Which workloads are affected

other

**Comment by andikrueger** [comment]
Thank you for raising this issue. Is this a new resource proposal or are these properties, that should be added to existing resources?

**Comment by AWeber78** [comment]
This is a new resource proposal

**Comment by andikrueger** [comment]
Are there any cmdlets / GraphAPIs available to manage these settings?

**Comment by AWeber78** [comment]
https://learn.microsoft.com/en-us/graph/api/resources/cloudpc-api-overview?view=graph-rest-beta

**Comment by AWeber78** [comment]
Needed Ressources:
- Provisioning Policy
- Custom Image
- Azure Network Connections (ANC)
- User Settings
- Cloud PC Maintenence Window
- Cloud PC encryption type

**Comment by j7lloyd** [comment]
@andikrueger, any update on this? Out of interest, what other pending information is required?

**Comment by AWeber78** [comment]
@andikrueger any update. What is still missing?

**Comment by AWeber78** [comment]
I have seen IntuneUserSettingsPolicyWindows365 is now part of the newest M365DSC Version.
Thanks for implementing

**Comment by FabienTschanz** [comment]
@AWeber78 Sure thing, there will be some more stuff coming in the next weeks. Don't always have the time to work on it, but I'll give my best shot. Keeping you posted. 

As a side note, some things probably won't be implemented because they require additional Azure things which are too "complex" to handle in a single resource. But I'll see what I can do.

Since you edited the message, I guess the issue with the not working export was resolved. I was surprised to get the notification that it doesn't because in my setup, it was working fine. Hope it now does for you as well 👍

**Comment by AWeber78** [comment]
@FabienTschanz 
Yes the issue was solved. Human error

I would not include the Azure / ANC part in M365DSC. This is process wise a prerequisite to a W365 Deployment.

**Comment by FabienTschanz** [comment]
@AWeber78 What do you exactly mean with `Cloud PC encryption type`? The Cloud PC maintenance window is still in preview and I won't include it yet until it's used in a broader context or out of preview. But I'll take add a note to myself and regularly check.

**Comment by AWeber78** [comment]
It’s unclear whether we made this setting available. I did not find anything. You can choose either a Microsoft Managed Key or a Customer Managed Key. The configuration status is visible within MSGraph.

https://graph.microsoft.com/beta/deviceManagement/virtualEndpoint/microsoft.graph.retrieveTenantEncryptionSetting()

**Comment by AWeber78** [comment]
The Alerts and Windows 365 roles are missing in the list

<img width="1326" height="559" alt="Image" src="https://github.com/user-attachments/assets/770c9c13-541b-426c-a8ce-f7bafab2e333" />

https://graph.microsoft.com/beta/deviceManagement/monitoring/alertRules

**Comment by FabienTschanz** [comment]
> It’s unclear whether we made this setting available. I did not find anything. You can choose either a Microsoft Managed Key or a Customer Managed Key. The configuration status is visible within MSGraph.
> 
> https://graph.microsoft.com/beta/deviceManagement/virtualEndpoint/microsoft.graph.retrieveTenantEncryptionSetting()

And how do I configure this setting? Unless it's a configuration that can be configured from the GUI, I will not implement it. And it also seems like there is no coverage in the Graph PowerShell module, and a public documentation about how to set this is also missing from the Microsoft docs. 

However, I'll take a look at implementing the alert rules.

**Comment by FabienTschanz** [comment]
The alert rules will be available with #6653. @AWeber78 What did you mean with Windows 365 roles? And do you have some news about the customer managed key?

**Comment by AWeber78** [comment]
Windows 365 implements another RBAC/role model in parallel with the existing one in Intune. This is not included in M365DSC, as is the case for Autopatch.

<img width="454" height="266" alt="Image" src="https://github.com/user-attachments/assets/e9376b3b-0195-4792-9d26-bdfb9491b2f8" />
 
The MS Graph Endpoint is https://graph.microsoft.com/beta/roleManagement/cloudPC/roleDefinitions

We are currently unable to implement the customer managed key. The interface is missing. This has to be on the to-do list for some time later.

**Comment by FabienTschanz** [comment]
Thanks, will take a look at those as well.

**Comment by FabienTschanz** [comment]
I have something prepared for Windows 365. Autopatch however cannot currently be implemented because it requires a Delegated authentication scenario, which is thus far not yet developed with the MSCloudLoginAssistant. I'm working on it too, but this has low priority.

**Comment by FabienTschanz** [comment]
@AWeber78 After some time, the PR #6873 will add the functionality for Windows365 roles. Cloud PC maintenance window is still in preview and thus not yet being implemented. But I have it on my todo list as soon as it's available.

**Comment by AWeber78** [comment]
Thank you for all your hard work. As the product evolves, some new configurations are missing. I will open a new issue for those that are not yet included.

**Comment by FabienTschanz** [comment]
Closing this one. Waiting for Cloud PC Maintenance Window to become GA.
